contrib/python/src/python/pants/contrib/python/checks/tasks/checkstyle/checkstyle.py

```diff
@@ -223,32 +225,47 @@ def execute(self):
     )
     if not python_tgts:
       return 0
+
     interpreter_cache = PythonInterpreterCache.global_instance()
+
     with self.invalidated(self.get_targets(self._is_checked)) as invalidation_check:
-      failure_count = 0
-      tgts_by_compatibility, _ = interpreter_cache.partition_targets_by_compatibility(
-        [vt.target for vt in invalidation_check.invalid_vts]
-      )
-      for filters, targets in tgts_by_compatibility.items():
+      targets_by_compatibility, _ = interpreter_cache.partition_targets_by_compatibility(
+        vt.target for vt in invalidation_check.invalid_vts)
+
+      # Accumulate all the sources by the minimum interpreter assigned to their owning target, and
+      # check them all at once to minimize the number of serial invocations of a checker pex.
+      # TODO: the v2 engine will allow parallelism in this task for free.
```

Comment:
Before reviewing - same question as on parent review - is this needed or are the lions share of the perf gains already realized by moving the pex out of .pants.d?